packages/fx-core/src/plugins/solution/fx-solution/utils/depsChecker/bicepChecker.ts

```diff
@@ -360,20 +367,17 @@ function getCommonProps(): { [key: string]: string } {
 async function displayLearnMore(
   message: string,
   link: string,
-  ctx: SolutionContext
+  ui?: UserInteraction,
+  telemetryReporter?: TelemetryReporter
 ): Promise<boolean> {
-  if (!ctx.ui) {
-    // no dialog, always continue
-    return true;
-  }
-  const res = await ctx.ui?.showMessage("info", message, true, Messages.learnMoreButtonText);
+  const res = await ui?.showMessage("info", message, true, Messages.learnMoreButtonText);
```

Comment:
It seems your refactor changes the original logic. When `ui === undefined`, the new logic will return `false` (and send telemetry) but the old logic will return `true`.

Reply:
fixed